docs/kubernetes/backup.md

```diff
@@ -0,0 +1,174 @@
+# Filestore Backups User Guide (Beta)
+
+>**Attention:** VolumeSnapshot is a Beta feature enabled by default in
+Kubernetes 1.17+.
+
+>**Attention:** VolumeSnapshot is only available in the driver version "master".
+
+### Backup Example
+A Filestore backup is a copy of a file share that includes all file data and metadata of the file share from the point in time when the backup is created. It works for Basic HDD and Basic SSD tier instances. Once a backup of a file share is created, the original file share can be modified or deleted without affecting the backup. A file share can be completely restored from a backup as a new Filestore instance or onto an existing file share. For more details refer to this [documentation](https://cloud.google.com/filestore/docs/backups)
+
+[CSI Snapshot](https://github.com/container-storage-interface/spec/blob/master/spec.md#createsnapshot) feature is leveraged to create Filestore Backups. By specifying a `type: backup` field in the VolumeSnapshotClass parameters, filestore CSI driver understands how to initiate a backup for a Filestore instance backed by the Persistent Volume.
+
+1. Create `StorageClass`
+
+    If you haven't created a `StorageClass` yet, create one first:
+
+    ```console
+    kubectl apply -f ./examples/kubernetes/backups-restore/sc.yaml
+    ```
+
+2. Create default `VolumeSnapshotClass`
+
+    ```console
+    kubectl create -f ./examples/kubernetes/backups-restore/backup-volumesnapshotclass.yaml
```

Comment:
+1 maybe add a precondition above to make sure of snapshot support & point to the k8s-csi documentation. And/or a plug for running on GKE where it will all just work :-)

Reply:
Added reference to the doc and GKE snapshot support.